src/Dscp/Crypto/Signing/Class.hs

```diff
@@ -56,6 +91,36 @@ deriving instance ByteArrayAccess (SK ss) =>
 deriving instance ByteArrayAccess (Sig ss) =>
     ByteArrayAccess (AbstractSig ss a)
 
+-- | Provide 'ByteArray' instances for signatures and keys.
+-- They are used for desirisalisation.
+deriving instance ByteArray (PK ss) =>
+    ByteArray (AbstractPK ss)
+deriving instance ByteArray (SK ss) =>
+    ByteArray (AbstractSK ss)
+deriving instance ByteArray (Sig ss) =>
+    ByteArray (AbstractSig ss a)
+
+-- | Provide 'FromByteArray' instances for signatures and keys.
+-- They are used for desirisalisation.
+deriving instance FromByteArray (PK ss) =>
+    FromByteArray (AbstractPK ss)
+deriving instance FromByteArray (SK ss) =>
+    FromByteArray (AbstractSK ss)
+deriving instance FromByteArray (Sig ss) =>
+    FromByteArray (AbstractSig ss a)
+
+-- | Generate a secret key.
+keyGen
```

Comment:
For ed25519 which we use as Signature scheme I believe, there is already a `generateSecretKey` function, http://hackage.haskell.org/package/cryptonite-0.25/docs/Crypto-PubKey-Ed25519.html. Can we make use of that one instead of creating our own here? For example having a `genSecretKey` function in our `SignatureScheme` class.

Reply:
Nice point, it seem to do exactly the same thing.